StreetPatch.Data/Repositories/CommentRepository.cs

```diff
@@ -12,5 +12,18 @@ public class CommentRepository : AbstractRepository<Comment, StreetPatchDbContex
         public CommentRepository(StreetPatchDbContext context) : base(context)
         {
         }
+
+        public async Task<Comment> DeleteOrArchiveAsync(Guid guid, bool isSoft)
+        {
+            if (!isSoft)
+            {
+                return await base.DeleteAsync(guid);
+            }
+
+            var comment = await base.GetAsync(guid);
+            comment.DeletedAt = DateTime.Now;
+            return await base.UpdateAsync(comment);
+
+        }
```

Comment:
You don't need this. We haven't specified that we are going to archive comments.